### PR TITLE
POC: Use reusable workflow

### PR DESCRIPTION
**Pull request by xzfc**
Issue: https://github.com/networkservicemesh/cmd-template/issues/95

For example usage, see https://github.com/networkservicemesh/sdk-sriov/pull/390.

### CODE REVIEW

File: .github/workflows/release-lib.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          git config --global user.email "nsmbot@networkservicmesh.io"
+          git config --global user.name "NSMBot"
+
+          release=${{ github.event.workflow_run.head_branch }}
+
+          for attempt in {1..5}; do
+            echo "attempt $attempt"
+            git fetch -p origin
+            base=$release
+            git show-ref --quiet refs/remotes/origin/$base || base=main
```
Comment (denis-tingaikin) [inline .github/workflows/release-lib.yaml:94]: note: base=main is not expected for release.

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

File: .github/workflows/release-lib.yaml (hunk 1 above)
Comment (xzfc) [reply .github/workflows/release-lib.yaml:94]: If the `$release` branch doesn't exist, we start it from the `main` branch. Or should we use a specific starting commit?

Comment (denis-tingaikin) [reply .github/workflows/release-lib.yaml:94]: I think we should stop release if we don't have release branch

Comment (xzfc) [reply .github/workflows/release-lib.yaml:94]: It's a branch in the dependent repository (not in the current one, which is running on the CI). Currently, branches in dependent repositories are created automatically from the `main` branch.

Details: our current `release.yaml` [uses](https://github.com/networkservicemesh/sdk-sriov/blob/9efc2a84b057be9ddfef30982ada07ee0e79e343/.github/workflows/release.yaml#L67-L72) the `actions/checkout` action to fetch the dependent repository. According to [`actions/checkout` doc](https://github.com/actions/checkout/blob/v2/README.md?plain=1#L45), the default (i.e. `main`) branch is used. Then our `release.yaml` [creates](https://github.com/networkservicemesh/sdk-sriov/blob/9efc2a84b057be9ddfef30982ada07ee0e79e343/.github/workflows/release.yaml#L90-L91) a release branch, dropping an already existing branch (since the force push `-f` flag being used).

---

In short, the current logic of `update-dependent-repositories` job:
Create a release branch starting from the `main` branch. Override it if it already exists (hence the conflict described in the issue).

Logic In this PR:
Create a release branch if it doesn't exist. Otherwise, append a commit to an existing release branch.
